parsl/executors/high_throughput/probe.py

```diff
@@ -0,0 +1,93 @@
+import zmq
+import argparse
+import uuid
+from parsl.addresses import get_all_addresses
+from zmq.utils.monitor import recv_monitor_message
+import time
+HEARTBEAT_CODE = (2 ** 32) - 1
+
+
+def probe_addresses(addresses, task_port, timeout=2):
+    """
+    Parameters
+    ----------
+
+    addresses: [string]
+        List of addresses as strings
+    task_port: int
+        Task port on the interchange
+    timeout: int
+        Timeout in seconds
+
+    Returns
+    -------
+    None or string address
+    """
+    context = zmq.Context()
+    addr_map = {}
+    for addr in addresses:
+        socket = context.socket(zmq.DEALER)
+        # socket.setsockopt(zmq.LINGER, 0)
+        url = f"tcp://{addr}:{task_port}"
+        print("Trying url: ", url)
+        socket.connect(url)
```

Comment:
does this wait for a tcp connection to be established/rejected before returning? if so, when the traffic is dropped rather than rejected (which is a fairly common firewall configuration on the internet) then this will maybe hang a-while?

Reply:
This is a zmq connect, and will return almost immediately. In fact it will buffer sends until a server is available remotely to receive.